core/src/test/java/org/frankframework/receivers/ConditionalPipelineCallerTest.java

```diff
@@ -0,0 +1,174 @@
+		CorePipeLineProcessor pipeLineProcessor = new CorePipeLineProcessor();
+		InputOutputPipeProcessor inputOutputPipeProcessor = new InputOutputPipeProcessor();
+		PipeProcessor pipeProcessor = new CorePipeProcessor();
+		inputOutputPipeProcessor.setPipeProcessor(pipeProcessor);
+
+		pipeLineProcessor.setPipeProcessor(inputOutputPipeProcessor);
+
+		PipeLine pl = spy(configuration.createBean(PipeLine.class));
+		pl.setFirstPipe("dummy");
+		pl.setPipeLineProcessor(pipeLineProcessor);
+
+		EchoPipe pipe = new EchoPipe() {
+			@Override
+			public PipeRunResult doPipe(Message message, PipeLineSession session) {
+				session.put(SESSION_VALUE_ECHO_PIPE_CALLED, "true");
+				return super.doPipe(message, session);
+			}
+		};
+		pipe.setName("dummy");
+		pl.addPipe(pipe);
+
+		PipeLineExit ple = new PipeLineExit();
+		ple.setName("success");
+		ple.setState(PipeLine.ExitState.SUCCESS);
+		pl.addPipeLineExit(ple);
+		adapter.setPipeLine(pl);
+
+		adapter.addReceiver(receiver);
+		configuration.addAdapter(adapter);
+		return adapter;
+	}
+
+	void startAdapter() throws ConfigurationException {
+		configuration.configure();
+		configuration.start();
+
+		waitForState(adapter, RunState.STARTED);
+		waitForState(receiver, RunState.STARTED);
+	}
+
+	@AfterEach
+	public void tearDown() throws Exception {
+		CloseUtils.closeSilently(session, configuration);
+	}
+
+	@Test
+	void testPipelineCaller() throws Exception {
+		// Arrange
+		String rawTestMessage = "TEST";
+		Message testMessage = new Message(new StringReader(rawTestMessage));
+
+		// Setup the echopipe to only run when the pipeline caller equals 'listener of [receiver]'
+		configureEchoPipe("listener of [receiver]");
+
+		// start adapter
+		startAdapter();
+
+		// Act
+		listener.processRequest(testMessage, session);
+
+		// Was set to 'true' in custom EchoPipe implementation in #setupAdapter
+		assertTrue(session.containsKey(SESSION_VALUE_ECHO_PIPE_CALLED));
+		assertEquals("true", session.get(SESSION_VALUE_ECHO_PIPE_CALLED));
+	}
+
+	@Test
+	void testPipelineCallerNoMatch() throws Exception {
+		// Arrange
+		String rawTestMessage = "TEST";
+		Message testMessage = new Message(new StringReader(rawTestMessage));
+
+		// This should never match with the listenerName
+		configureEchoPipe("niet goed");
+
+		// start adapter
+		startAdapter();
+
+		// Act
+		listener.processRequest(testMessage, session);
+
+		// Was not set to 'true' in custom EchoPipe implementation in #setupAdapter.
+		assertTrue(session.containsKey(SESSION_VALUE_ECHO_PIPE_CALLED));
+		assertNull(session.get(SESSION_VALUE_ECHO_PIPE_CALLED));
+	}
+
+	private void configureEchoPipe(String ifValue) throws ConfigurationException {
+		EchoPipe echoPipe = (EchoPipe) adapter.getPipeLine().getPipe("dummy");
+		echoPipe.setIfParam("paramInput");
+		echoPipe.setIfValue(ifValue);
+
+		Parameter parameter = new Parameter();
+		parameter.setName("paramInput");
+		parameter.setContextKey(Receiver.CONTEXT_PIPELINE_CALLER);
```

Comment:
Het is handiger als deze twee context keys bij elkaar in de MessageContext zitten.